sdk/appconfiguration/azure-data-appconfiguration/src/configuration_client.cpp

```diff
@@ -0,0 +1,2316 @@
+{
+  auto const pageToken = this->NextPageToken;
+  this->m_options.NextPageToken = pageToken.Value();
+  *this = this->m_client->GetKeys(this->m_accept, this->m_options, context);
+  this->CurrentPageToken = pageToken.Value();
+}
+
+void GetKeyValuesPagedResponse::OnNextPage(Core::Context const& context)
+{
+  auto const pageToken = this->NextPageToken;
+  this->m_options.NextPageToken = pageToken.Value();
+  *this = this->m_client->GetKeyValues(this->m_accept, this->m_options, context);
+  this->CurrentPageToken = pageToken.Value();
+}
+
+void GetSnapshotsPagedResponse::OnNextPage(Core::Context const& context)
+{
+  auto const pageToken = this->NextPageToken;
+  this->m_options.NextPageToken = pageToken.Value();
+  *this = this->m_client->GetSnapshots(this->m_accept, this->m_options, context);
+  this->CurrentPageToken = pageToken.Value();
+}
+
+void GetLabelsPagedResponse::OnNextPage(Core::Context const& context)
+{
+  auto const pageToken = this->NextPageToken;
+  this->m_options.NextPageToken = pageToken.Value();
+  *this = this->m_client->GetLabels(this->m_accept, this->m_options, context);
+  this->CurrentPageToken = pageToken.Value();
+}
+
+void GetRevisionsPagedResponse::OnNextPage(Core::Context const& context)
+{
+  auto const pageToken = this->NextPageToken;
+  this->m_options.NextPageToken = pageToken.Value();
+  *this = this->m_client->GetRevisions(this->m_accept, this->m_options, context);
+  this->CurrentPageToken = pageToken.Value();
+}
+
+ConfigurationClient::ConfigurationClient(
+    std::string const& url,
+    std::shared_ptr<Core::Credentials::TokenCredential> const& credential,
+    ConfigurationClientOptions options)
+    : m_url(url), m_apiVersion(options.ApiVersion)
+{
+  std::vector<std::unique_ptr<Core::Http::Policies::HttpPolicy>> perRetryPolicies;
+  {
+    Core::Credentials::TokenRequestContext tokenRequestContext;
+    tokenRequestContext.Scopes = {"https://azconfig.io/.default"};
+    perRetryPolicies.emplace_back(
+        std::make_unique<Core::Http::Policies::_internal::BearerTokenAuthenticationPolicy>(
+            credential, tokenRequestContext));
+  }
+
+  m_pipeline.reset(new Core::Http::_internal::HttpPipeline(
+      {}, "data-appconfiguration", PackageVersion::ToString(), std::move(perRetryPolicies), {}));
+}
+
+std::string ConfigurationClient::GetUrl() const { return m_url.GetAbsoluteUrl(); }
+
+GetKeysPagedResponse ConfigurationClient::GetKeys(
+    std::string const& accept,
+    GetKeysOptions const& options,
+    Core::Context const& context)
+{
+  auto url = m_url;
+  url.AppendPath("keys");
+
+  url.AppendQueryParameter("api-version", Core::Url::Encode(m_apiVersion));
+  url.AppendQueryParameter("name", Core::Url::Encode(options.Name));
+  url.AppendQueryParameter("After", Core::Url::Encode(options.After));
+  if (options.NextPageToken.find("https://") == 0 || options.NextPageToken.find("http://") == 0)
+  {
+    url = Core::Url(options.NextPageToken);
```

Comment:
We'll want to review the implementation detail as part of the PagedResponse issue:

That said, regarding naming our existing PagedResponse type does call it a NextPageToken, instead of url.
https://github.com/Azure/azure-sdk-for-cpp/blob/3687136d04cc8e425cb368ce879a1686834b17c9/sdk/keyvault/azure-security-keyvault-certificates/src/certificate_client.cpp#L56-L62

https://github.com/Azure/azure-sdk-for-cpp/blob/3687136d04cc8e425cb368ce879a1686834b17c9/sdk/core/azure-core/inc/azure/core/paged_response.hpp#L63-L78